src/fft.jl

```diff
@@ -640,10 +643,10 @@ for (Tr,Tc,fftw,lib) in ((:Float64,:(Complex{Float64}),"fftw",:libfftw3),
         return rFFTWPlan{$Tc,$BACKWARD,inplace,N}(plan, flags, R, X, Y)
     end
 
-    @eval @exclusive function r2rFFTWPlan{$Tr,Any,inplace,N}(X::StridedArray{$Tr,N},
-                                                  Y::StridedArray{$Tr,N},
-                                                  region, kinds, flags::Integer,
-                                                  timelimit::Real) where {inplace,N}
+    @eval @inline function _r2rFFTWPlan(X::StridedArray{$Tr,N},
+                          Y::StridedArray{$Tr,N},
+                          region, kinds, flags::Integer,
+                          timelimit::Real, ::Val{inplace}) where {inplace,N}
         R = isa(region, Tuple) ? region : copy(region)
         knd = fix_kinds(region, kinds)
```

Comment:
Couldn't all of this logic be put into `fix_kinds` instead, to ensure that it produces a type-stable tuple?

Reply:
On further reflection, do we need to store the value of kinds as a type parameter? It doesn't seem to be used anywhere other than `show`, which isn't that important anyway